docs/standard/serialization/system-text-json-how-to.md

```diff
@@ -0,0 +1,768 @@
+
+## How to deserialize
+
+Call [JsonSerializer.Deserialize](xref:System.Text.Json.JsonSerializer.Deserialize*):
+
+```csharp
+var weatherForecast = JsonSerializer.Deserialize<WeatherForecast>(json);
+```
+
+Example JSON input:
+
+```json
+{
+  "Date": "2019-08-01T00:00:00-07:00",
+  "temperatureC": 25,
+  "Summary": "Hot",
+  "DatesAvailable": [
+    "2019-08-01T00:00:00-07:00",
+    "2019-08-02T00:00:00-07:00"
+  ],
+  "TemperatureRanges": {
+    "Cold": {
+      "High": {
+        "DegreesCelsius": 20
+      },
+      "Low": {
+        "DegreesCelsius": -10
+      }
+    },
+    "Hot": {
+      "High": {
+        "DegreesCelsius": 60
+      },
+      "Low": {
+        "DegreesCelsius": 20
+      }
+    }
+  },
+  "SummaryWords": [
+    "Cool",
+    "Windy",
+    "Humid"
+  ]
+}
+```
+
+Resulting property and field values:
+
+|Property |Value  |Notes  |
+|---------|---------|---------|
+| Date    | 8/1/2019 12:00:00 AM -07:00||
+| TemperatureC| 0 | Not set because property name matching is case-sensitive.|
+| Summary| Hot||
+| SummaryField| null | Not set because it's a field, not a property.|
+| DatesAvailable | 8/1/2019 12:00:00 AM -07:00<br>8/2/2019 12:00:00 AM -07:00 | `IList<T>` and `IEnumerable<T>` implementations are supported. |
+| TemperatureRanges | Cold, 20 High -10 Low<br>Hot, 60 High 20 Low| `Dictionary<string,TValue>` is supported. |
+| SummaryWords | Cool<br>Windy<br>Humid | Arrays are supported. |
+
+Overloads of <xref:System.Text.Json.JsonSerializer.Deserialize*> let you deserialize from a `Stream`.  Async versions of the `Stream` overloads are available.
+
+### Default deserialization behavior
+
+* If the JSON contains a value for a read-only property, the value is ignored and no exception is thrown.
+* Deserialization to reference types without a parameterless constructor isn't supported.
+* Deserialization to [immutable objects](https://github.com/dotnet/corefx/issues/38569) or [read-only properties](https://github.com/dotnet/corefx/issues/38163) isn't supported.
+* Comments or trailing commas in the JSON trigger exceptions.
+* A nullable property is set to null if the JSON has a null value or the property is missing from the JSON. For information about how to differentiate these two situations, see [GitHub issue 37485](https://github.com/dotnet/corefx/issues/37485).
+
+## Serialize to formatted JSON
+
+Set <xref:System.Text.Json.JsonSerializerOptions.WriteIndented> to true:
+
+```csharp
+var options = new JsonSerializerOptions
+{
+    WriteIndented = true,
+};
+json = JsonSerializer.Serialize(weatherForecast, options);
+```
+
+Example type to be serialized and JSON output:
+
+```csharp
+class WeatherForecast
+{
+    public DateTimeOffset Date { get; set; }
+    public int TemperatureC { get; set; }
+    public string Summary { get; set; }
+}
+```
+
+```json
+{
+  "Date": "2019-08-01T00:00:00-07:00",
+  "TemperatureC": 25,
+  "Summary": "Hot"
+}
+```
+
+## Allow comments and trailing commas
```

Comment:
Presumably, some of these samples should make their way into the API reference directly as well (and not just in this concept docs). Do you agree? Do we have plans to do that?

For example, in here:
https://docs.microsoft.com/en-us/dotnet/api/system.text.json.jsonserializeroptions.allowtrailingcommas?view=netcore-3.0

Speaking of which, @mairaw, @rpetrusha - another remarks formatting issue:
![image](https://user-images.githubusercontent.com/6527137/64854650-fc0bf000-d5d2-11e9-90ea-d2a1fda77424.png)

https://github.com/dotnet/dotnet-api-docs/blob/470813feff2d3dd5c79c4ee3a7072417ebf1da11/xml/System.Text.Json/JsonSerializerOptions.xml#L57-L65

Is there a validating tool we can run over all the remarks section and make sure they don't contain the "exception" tags?

cc @carlossanlop

Reply:
>some of these samples should make their way into the API reference directly as well (and not just in this concept docs). Do you agree? Do we have plans to do that?

I agree.  I'll  update related API ref once the content in the conceptual docs is approved.